packages/salesforcedx-utils-vscode/src/util/authInfo.ts

```diff
@@ -152,6 +152,32 @@ export const getConnection = async (usernameOrAlias?: string): Promise<Connectio
 /** Get auth fields for a connection */
 export const getAuthFields = (connection: Connection): AuthFields => connection.getAuthInfoFields();
 
+/**
+ * Gets the org API version as a numeric value.
+ * Uses instanceApiVersion (max supported API version) from auth fields if available,
+ * otherwise falls back to the connection's configured API version.
+ * @returns The org API version as a number, or undefined if unable to retrieve.
+ */
+export const getOrgApiVersion = async (): Promise<number | undefined> => {
```

Comment:
I feel like we already have something like this in so many different places! Can you seek to consolidate it somewhere? 
- https://github.com/forcedotcom/salesforcedx-vscode/blob/76402aa0872218f3993e8eb7f444654740b2fbe8/packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts#L62
- https://github.com/forcedotcom/salesforcedx-vscode/blob/66dfb7dbb74e89b9f9b9a9ea0e11ba5646604694/packages/salesforcedx-vscode-apex-oas/src/oas/generationStrategy/json/auraEnabledStrategy.ts#L102

Not sure if they're the same though